From: Robert Schwebel <r.schwebel@pengutronix.de>
Date: Wed, 23 Sep 2026 16:36:40 +0200
Subject: [PATCH] libtheora: fix configure.ac to work with cross scenario for
 sdl

This is https://trac.xiph.org/ticket/1466

The current test in configure.ac mixes up two different methods (hand
written plus pkg-config). This patch changes the mechanics towards
proper pkg-config usage. In return, we get a cleaned up cross scenario
with no build-system leakage.

This patch fixes the SDL test.

Signed-off-by: Robert Schwebel <r.schwebel@pengutronix.de>
---
 configure.ac | 28 ++++++++++++++++++++++------
 1 file changed, 22 insertions(+), 6 deletions(-)

diff --git a/configure.ac b/configure.ac
index 234acd501685..8a4faa7d344f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -467,12 +467,28 @@ if test "${HAVE_VORBIS}" = "yes"; then
 fi
 
 dnl check for SDL
-HAVE_SDL=no
-if test "x$HAVE_PKG_CONFIG" = "xyes"
-then
-  PKG_CHECK_MODULES(SDL, sdl, HAVE_SDL=yes, [
-    HAVE_SDL=no
-    AC_MSG_WARN([*** Unable to find SDL -- Not compiling example players ***])])
+AC_MSG_CHECKING([for sdl])
+AC_ARG_ENABLE(sdl,
+    AS_HELP_STRING([--enable-sdl], [enable SDL @<:@default=yes@:>@]),
+        [case "$enableval" in
+        y | yes) HAVE_SDL=yes ;;
+        *) HAVE_SDL=no ;;
+        esac],
+    [HAVE_SDL=yes])
+AC_MSG_RESULT([${HAVE_SDL}])
+AM_CONDITIONAL(HAVE_SDL, [test "$HAVE_SDL" = "yes"])
+if test "${HAVE_SDL}" = "yes"; then
+    AC_DEFINE(HAVE_SDL, 1, [sdl])
+
+    REQUIRES_SDL="sdl >= 1.2.13"
+    AC_SUBST(REQUIRES_SDL)
+    PKG_CHECK_MODULES([SDL],
+          [${REQUIRES_SDL}],
+          [],
+          [AC_MSG_ERROR([*** ${REQUIRES_SDL} not found by pkg-config on your system])]
+    )
+    AC_SUBST(SDL_CFLAGS)
+    AC_SUBST(SDL_LIBS)
 fi
 
 dnl check for OSS
